core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.ts

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for image uploader modal.
+ */
+
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild,
+} from '@angular/core';
+import {SafeResourceUrl} from '@angular/platform-browser';
+import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
+import {ConfirmOrCancelModal} from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import Cropper from 'cropperjs';
+import {SvgSanitizerService} from 'services/svg-sanitizer.service';
+import {WindowDimensionsService} from 'services/contextual/window-dimensions.service';
+require('cropperjs/dist/cropper.min.css');
+
+@Component({
+  selector: 'oppia-image-uploader-modal',
+  templateUrl: './image-uploader-modal.component.html',
+})
+export class ImageUploaderModalComponent extends ConfirmOrCancelModal {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() allowedImageFormats!: string[];
+  @Input() bgColor!: string;
+  @Input() previewFooter!: string;
+  @Input() previewDescription!: string;
+  @Input() previewTitle!: string;
+  @Input() previewDescriptionBgColor!: string;
+  @Input() imageName!: string;
+  @Input() aspectRatio!: string;
+  @Input() maxImageSizeInKB!: number;
+  @Input() previewImageUrl!: string;
+
+  // 'uploadedImage' will be null if the uploaded svg is invalid or not trusted.
+  uploadedImage: SafeResourceUrl | null = null;
+  thumbnailImageDataUrl: string | null = null;
+  croppedImageDataUrl: string | SafeResourceUrl | null = null;
```

Comment:
Shouldn't this be one Url? It seems like you are paying too much attention to the thumbnails. I understand they need special treatment, but it'd be much better if there was only one url. What if there's a third type of image in the future, we sure don't want to add another Url field

Reply:
make sense. removed it and now we only use uploadedImage to display preview.